lms/static/scripts/frontend_apps/errors.js

```diff
@@ -105,3 +118,46 @@ export function isLTILaunchServerError(error) {
     ].includes(error.errorCode)
   );
 }
+
+/**
+ * Format a user-facing message based on this error and optional contextual
+ * prefix, using the appropriate message information.
+ *
+ * @param {ErrorLike} error
+ * @param {String} [prefix]
+ * @returns {String}
+ */
+export function formatErrorMessage(error, prefix = '') {
+  // If `serverMessage` is extant on `error`, prefer it to `error.message` even
+  // if `serverMessage` is empty . In cases where we are displaying error
+  // information provided by the backend (i.e. `APIError`), we do not want
+  // to render the JS Error instance's `message` as it likely does not apply
```

Comment:
"likely does not apply" seems unclear to me. In the case of `APIError` the `message` property "does apply", but is a generic error whereas `serverMessage` is more suitable for display to an end user. Perhaps revise to something like "If an error message is provided by the backend (eg. in `APIError`) it is preferred over `message` for display to end users even if empty."

Reply:
Revising entire comment to:

```
  // If any message is provided by the backend as `error.serverMessage`,
  // prefer this for display to users even if it is empty.
```